### PR TITLE
[MI-3559] Add batching in APIs in various client functions

### PR DESCRIPTION
**Pull request by ayusht2810**
#### Summary
Add batching in APIs in various client functions

#### Ticket Link
#331

### CODE REVIEW

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/message_hooks.go [hunk 1]
```diff
@@ -652,31 +640,32 @@ func (p *Plugin) Update(teamID, channelID string, user *model.User, newPost, old
 		return errors.New("post not found")
 	}
 
+	var updatedMessage *msteams.Message
 	if updateRequired {
 		// TODO: Add the logic of processing the attachments and uploading new files to Teams
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply server/message_hooks.go:645]: discussed

File: server/msteams/client.go [hunk 2]
```diff
@@ -683,19 +684,48 @@ func (tc *ClientImpl) UpdateMessage(teamID, channelID, parentID, msgID, message
 	body.SetContent(&message)
 	rmsg.SetBody(body)
 
+	var updateMessageRequest *abstractions.RequestInformation
 	if parentID != "" {
-		if _, err := tc.client.Teams().ByTeamId(teamID).Channels().ByChannelId(channelID).Messages().ByChatMessageId(parentID).Replies().ByChatMessageId1(msgID).Patch(tc.ctx, rmsg, nil); err != nil {
-			return NormalizeGraphAPIError(err)
+		updateMessageRequest, err = tc.client.Teams().ByTeamId(teamID).Channels().ByChannelId(channelID).Messages().ByChatMessageId(parentID).Replies().ByChatMessageId1(msgID).ToPatchRequestInformation(tc.ctx, rmsg, nil)
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply server/msteams/client.go:689]: discussed

File: server/msteams/client.go [hunk 3]
```diff
@@ -721,11 +751,29 @@ func (tc *ClientImpl) UpdateChatMessage(chatID, msgID, message string, mentions
...
 	}
 
-	return nil
+	getMessageRequest, err := tc.client.Chats().ByChatId(chatID).Messages().ByChatMessageId(msgID).ToGetRequestInformation(tc.ctx, nil)
+	if err != nil {
+		return nil, NormalizeGraphAPIError(err)
+	}
+
+	batchRequest := msgraphcore.NewBatchRequest(tc.client.GetAdapter())
+	setReactionRequestItem, err := batchRequest.AddBatchRequestStep(*updateMessageRequest)
```
Comment (manojmalik20) [inline server/msteams/client.go:765]: Variable name is wrong here. It should be `updateMessageRequestItem`.

File: server/msteams/client.go [hunk 4]
```diff
@@ -1428,37 +1476,89 @@ func (tc *ClientImpl) SetChatReaction(chatID, messageID, userID, emoji string) e
...
+		setReactionRequest, err = tc.client.Teams().ByTeamId(teamID).Channels().ByChannelId(channelID).Messages().ByChatMessageId(parentID).Replies().ByChatMessageId1(messageID).SetReaction().ToPostRequestInformation(tc.ctx, setReaction, nil)
+		if err != nil {
+			return nil, NormalizeGraphAPIError(err)
 		}
 	}
-	return nil
+
+	var getMessageRequest *abstractions.RequestInformation
+	if parentID != "" {
+		getMessageRequest, err = tc.client.Teams().ByTeamId(teamID).Channels().ByChannelId(channelID).Messages().ByChatMessageId(messageID).Replies().ByChatMessageId1(parentID).ToGetRequestInformation(tc.ctx, nil)
```
Comment (manojmalik20) [inline server/msteams/client.go:1535]:
```suggestion
		getMessageRequest, err = tc.client.Teams().ByTeamId(teamID).Channels().ByChannelId(channelID).Messages().ByChatMessageId(parentID).Replies().ByChatMessageId1(messageID).ToGetRequestInformation(tc.ctx, nil)
```
Check this everywhere.

File: server/msteams/client.go [hunk 5]
```diff
@@ -1469,34 +1569,86 @@ func (tc *ClientImpl) UnsetChatReaction(chatID, messageID, userID, emoji string)
...
+		unsetReactionRequest, err = tc.client.Teams().ByTeamId(teamID).Channels().ByChannelId(channelID).Messages().ByChatMessageId(parentID).Replies().ByChatMessageId1(messageID).UnsetReaction().ToPostRequestInformation(tc.ctx, unsetReaction, nil)
+		if err != nil {
+			return nil, NormalizeGraphAPIError(err)
 		}
 	}
-	return nil
+
+	var getMessageRequest *abstractions.RequestInformation
+	if parentID != "" {
+		getMessageRequest, err = tc.client.Teams().ByTeamId(teamID).Channels().ByChannelId(channelID).Messages().ByChatMessageId(messageID).Replies().ByChatMessageId1(parentID).ToGetRequestInformation(tc.ctx, nil)
```
Comment (manojmalik20) [inline server/msteams/client.go:1628]:
```suggestion
		getMessageRequest, err = tc.client.Teams().ByTeamId(teamID).Channels().ByChannelId(channelID).Messages().ByChatMessageId(parentID).Replies().ByChatMessageId1(messageID).ToGetRequestInformation(tc.ctx, nil)
```

File: server/message_hooks_test.go [hunk 6]
```diff
@@ -362,6 +321,16 @@ func TestMessageHasBeenUpdated(t *testing.T) {
 			},
 		},
 	}
+	mockChannelMessage := &msteams.Message{
+		ID:        "mockMessageID",
+		TeamID:    "mockTeamsTeamID",
+		ChannelID: "mockTeamsChannelID",
+		ChatID:    testutils.GetChatID(),
```
Comment (manojmalik20) [inline server/message_hooks_test.go:328]: No need for keeping chat ID in a channel message.

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**